libs/langchain/langchain/vectorstores/milvus.py

```diff
@@ -821,3 +821,87 @@ def from_texts(
         )
         vector_db.add_texts(texts=texts, metadatas=metadatas)
         return vector_db
+
+    def get_pks(self, expr: str = None, **kwargs: Any) -> List[int] | None:
+        """ Get pks (primary keys) with expression
+
+        Args:
+            expr: Expression - E.g: "id in [1, 2]", or "title LIKE 'Abc%'"
+        
+        Returns:
+            List[int]: List of PKs (Primary Keys)
+        """
+
+        from pymilvus import MilvusException
+
+        if self.col is None:
+            logger.debug("No existing collection to get pk.")
+            return False
+        
+        try:
+            query_result = self.col.query(
+                    expr=expr,
+                    output_fields=["pk"]
+                )
+        except MilvusException as exc:
+            logger.error(
+                "Failed to get pks: %s error: %s", self.collection_name, exc
+            )
+            raise exc
+        pks = [item.get("pk") for item in query_result]
+        
+        return pks
+ 
+    def delete(self, pks: List[int] | None = None, **kwargs: Any) -> bool | None:
```

Comment:
can we use the method signature inherited from parent class?
```python
    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> Optional[bool]:
```

Reply:
Yes, I’ve updated it. You can check it now